src/thread.ts

```diff
@@ -140,226 +146,200 @@ export class Thread<SCG extends ExtendableGenerics = DefaultGenerics> {
 
-    this.unsubscribeFunctions.add(
-      this.client.on('message.read', (event) => {
-        if (!event.user || !event.created_at || !event.thread) return;
-        if (event.thread.parent_message_id !== this.id) return;
+  private subscribeRepliesRead = () =>
+    this.client.on('message.read', (event) => {
+      if (!event.user || !event.created_at || !event.thread) return;
+      if (event.thread.parent_message_id !== this.id) return;
 
-        const userId = event.user.id;
-        const createdAt = event.created_at;
-        const user = event.user;
+      const userId = event.user.id;
+      const createdAt = event.created_at;
+      const user = event.user;
 
-        // FIXME: not sure if this is correct at all
-        this.state.next((current) => ({
-          ...current,
-          read: {
-            ...current.read,
-            [userId]: {
-              last_read: createdAt,
-              lastReadAt: new Date(createdAt),
-              user,
-              // TODO: rename all of these since it's formatted (find out where it's being used in the SDK)
-              unread_messages: 0,
-              // TODO: fix this (lastestReplies.at(-1) might include message that is still being sent, which is wrong)
-              last_read_message_id: 'unknown',
-            },
+      this.state.next((current) => ({
+        ...current,
+        read: {
+          ...current.read,
+          [userId]: {
+            lastReadAt: new Date(createdAt),
+            user,
+            lastReadMessageId: event.last_read_message_id,
+            unreadMessageCount: 0,
           },
-        }));
-      }).unsubscribe,
-    );
-
-    const handleMessageUpdate = (event: Event<SCG>) => {
-      if (!event.message) return;
+        },
+      }));
+    }).unsubscribe;
 
-      if (event.hard_delete && event.type === 'message.deleted' && event.message.parent_id === this.id) {
+  private subscribeReplyDeleted = () =>
+    this.client.on('message.deleted', (event) => {
+      if (event.message?.parent_id === this.id && event.hard_delete) {
         return this.deleteReplyLocally({ message: event.message });
       }
+    }).unsubscribe;
+
+  private subscribeMessageUpdated = () => {
+    const unsubscribeFunctions = ['message.updated', 'reaction.new', 'reaction.deleted'].map(
+      (eventType) =>
+        this.client.on(eventType, (event) => {
+          if (event.message) {
+            this.updateParentMessageOrReplyLocally(event.message);
+          }
+        }).unsubscribe,
+    );
 
-      this.updateParentMessageOrReplyLocally(event.message);
-    };
-
-    ['message.updated', 'message.deleted', 'reaction.new', 'reaction.deleted'].forEach((eventType) => {
-      this.unsubscribeFunctions.add(this.client.on(eventType, handleMessageUpdate).unsubscribe);
-    });
+    return () => unsubscribeFunctions.forEach((unsubscribe) => unsubscribe());
   };
 
-  public deregisterSubscriptions = () => {
+  public unregisterSubscriptions = () => {
```

Comment:
I was following [this post](https://english.stackexchange.com/a/40095) on how to name this one, I'm fine with either though (we user `subscribe`/`unsubscribe` anyway).

Reply:
I'm trying to remember why it was renamed - I think for consistency with `unsubscribe` somewhere else :) I'll give it another thought before merging.

Reply:
Let's keep unregister for consistency, I think it's fine. :)